_posts/2024-12-30-debezium-oracledb-logminer.md

```diff
@@ -0,0 +1,313 @@
+        - ./oracleDB/oradata:/opt/oracle/oradata
+  
+  4.  Create directory “oracleDB/oradata and change ownership to 54321
+        '''
+        $> mkdir -p oracleDB/oradata
+        $> sudo chown -R oracleDB/oradata 54321:54321
+        '''
+  5.  Start Confluent Platform stack in detach mode:
+        
+        '''
+        $> docker compose up -d
+          Each component of Confluent Platform starts in separate container.
+          
+        Creating broker   ... done
+        Creating schema-registry  ... done
+        Creating rest-proxy       ... done
+        Creating connect          ... done
+        Creating ksqldb-server    ... done
+        Creating control-center   ... done
+        Creating ksql-datagen     ... done
+        Creating ksqldb-cli       ... done
+        Creating oracle           ... done
+        
+        '''
+
+        You can verify if all services are up and running using command “docker compose ps -a”
+
+# Configuring Oracle Database
+
+  When Oracle DB container starts for the first time, there are no initial configuration and database exists. Hence database will be installed and configuration will be started. This process may take 10-20 mins (approx). You can verify readiness of Oracle DB from container logs (docker compose logs oracle) and should see message:
+
+    '''
+    #############################
+    DATABASE IS READY TO USE!
+    #############################
+    '''
+
+
+  In order to incorporate changes from an Oracle database, a number of database configurations are required:
+
+
+# Enable Archive logs
+
+  The Oracle container registry image used in the Install Oracle section may not have archive logging enabled. If you use another image or a pre-existing environment, you should check whether archive logging is enabled.
+
+  '''
+  $> docker compose exec oracle bash
+  $> sqlplus ‘/ as sysdba’
+
+  SQL> SELECT LOG_MODE FROM V$DATABASE
+
+  '''
+
+  If the column contains ARCHIVELOG, then archive logging is enabled. If the column contains the value NOARCHIVELOG, archive logging isn’t enabled, and further configuration is necessary.
+
+  Execute the following SQL commands inside the SQL*Plus terminal window:
+  
+  '''
+  ALTER SYSTEM SET db_recovery_file_dest_size = 10G;
+  ALTER SYSTEM SET db_recovery_file_dest = '/opt/oracle/oradata/ORCLCDB' scope=spfile;
+  SHUTDOWN IMMEDIATE
+  STARTUP MOUNT
+  ALTER DATABASE ARCHIVELOG;
+  ALTER DATABASE OPEN;
+  ARCHIVE LOG LIST;
+  '''
+  Output of last executed SQL command should show Archive mode for Database log mode.
+  '''
+  Database log mode                  Archive Mode
+  Automatic archival                 Enabled
+  Archive destination                USE_DB_RECOVERY_FILE_DEST
+  Oldest online log sequence         1
+  Next log sequence to archive       3
+  Current log sequence               3
+  '''
+
+
+
+# Redo Logs
+
+  Oracle Redo logs are the transactional logs. Using the same terminal window, execute below listed SQL command:
```

Comment:
Does this description convey the necessary details about the commands coming up next?